internal/blobstore/blobstore.go

```diff
@@ -76,22 +76,65 @@ func NewContentChallengeResponse(chal *ContentChallenge, r io.ReadSeeker) (*Cont
 	}, nil
 }
 
+// Store stores data blobs in the configured store.
+type Store struct {
+	store
+}
+
 // Store stores data blobs in mongodb, de-duplicating by
 // blob hash.
-type Store struct {
+type store interface {
+	Put(r io.Reader, name string, size int64, hash string, proof *ContentChallengeResponse) (*ContentChallenge, error)
```

Comment:
Also, I'd like to see more doc comments on the methods here to make their contracts clear.